src/lib/Lib/CMakeExecution.cpp

```diff
@@ -104,8 +131,55 @@ getCmakeDefaultGenerator(llvm::StringRef cmakePath)
     return Unexpected(Error("Default CMake generator not found"));
 }
 
+Expected<std::string>
+parseCmakeSystemInformationOutput(std::string const& cmakeSystemInformation)
+{
+    std::istringstream stream(cmakeSystemInformation);
+    std::string line;
+    std::string defaultGenerator;
+
+    while (std::getline(stream, line))
+    {
+        if (line.starts_with("CMAKE_GENERATOR \""))
+        {
+            size_t const start = 17;
+            size_t const end = line.find("\"", start);
+            if (end == std::string::npos)
+            {
+                continue;
+            }
+            return line.substr(start, end - start);
+        }
+    }
+    return Unexpected(Error("Default CMake generator not found"));
+}
+
+Expected<std::string>
+getCmakeDefaultGenerator(llvm::StringRef cmakePath)
+{
+    Expected<std::string> const cmakeHelpExp = executeCmakeHelp(cmakePath);
+    if (cmakeHelpExp)
+    {
+        std::string const cmakeHelp = *std::move(cmakeHelpExp);
+        return parseCmakeHelpOutput(cmakeHelp);
```

Comment:
The code looks cleaner this way but now I'm wondering: executeCmakeHelp doesn't fail but parseCmakeHelpOutput does, shouldn't we move on and fall back to executeCmakeSystemInformation? And of course the same logic would apply to executeCmakeSystemInformation.

Reply:
```cpp
    Expected<std::string> const cmakeHelpExp = executeCmakeHelp(cmakePath);
    if (cmakeHelpExp)
    {
        std::string const cmakeHelp = *std::move(cmakeHelpExp);
        auto r = parseCmakeHelpOutput(cmakeHelp);
        if (r) 
        {
            return *r;
        }
    }
    // ... let it fall back to the next case...
```

Reply:
Good suggestion